tokio/src/time/driver/wheel/mod.rs

```diff
@@ -125,8 +137,12 @@ impl Wheel {
     }
 
     /// Advances the timer up to the instant represented by `now`.
-    pub(crate) fn poll(&mut self, now: u64) -> Option<OwnedItem> {
+    pub(crate) fn poll(&mut self, now: u64) -> Option<TimerHandle> {
```

Comment:
It looks like this fn conflates processing the timer (moving entries down levels and finding entries that are expired) with popping the entry back in order to wake it. Maybe that could be split up. `wheel::process()` would move all expired to the pending list. Then `WHeel::pop_pending()` would remove the entries.